src/components/shared/table.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+function Table({ data, columns, editable, editUrl }) {
+  const renderData = (data, columns) => {
+    return data.map((row) => (
+      <tr key={row.id}>
+        {columns.map((column) => (
+          <td key={column.name}>
+            {editable && column.name === 'editable' ? (
+              <Link to={editUrl}>Edit</Link>
+            ) : (
+              row[column.name]
+            )}
+          </td>
+        ))}
+      </tr>
+    ));
+  };
+  return (
+    <div>
+      <table className="table">
+        <thead>
+          <tr>
+            {columns.map((heading) => (
+              <th scope="col" key={heading.name}>
+                {heading.header}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>{data && renderData(data, columns)}</tbody>
```

Comment:
If you used a renderRow method and checked for if there is data in there you could move this display logic and have a simple solution for a loading/empty state.